### PR TITLE
Move logic to the Notice Object.

### PR DESCRIPTION
**Pull request by Camwyn**
Split template into twins based on the logic.

https://www.loom.com/share/d0fab2d2aca0410b88546f4ccab46156?sid=9fbb9d0b-9cbc-47a9-904c-9e6ee453fbe1
[TEC-4841]

[TEC-4841]: https://theeventscalendar.atlassian.net/browse/TEC-4841?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by stratease** [comment]
Could use a simple test?

File: src/Tribe/Admin/Notice/Marketing/Stellar_Sale.php [hunk 1]
```diff
@@ -66,10 +66,14 @@ public function display_notice() {
...
+		$dependency   = tribe( \Tribe__Dependency::class );
 
 		ob_start();
-
-		include \Tribe__Main::instance()->plugin_path . 'src/admin-views/notices/tribe-stellar-sale.php';
+		if ( $dependency->has_active_premium_plugin() ) {
+			include \Tribe__Main::instance()->plugin_path . 'src/admin-views/notices/tribe-stellar-sale-premium.php';
+		} else {
+			include \Tribe__Main::instance()->plugin_path . 'src/admin-views/notices/tribe-stellar-sale.php';
+		}
```
Comment (bordoni) [inline src/Tribe/Admin/Notice/Marketing/Stellar_Sale.php:76]: WE should use `Tribe__Template` honestly

**Review by bordoni — CHANGES_REQUESTED** [review]

**Comment by EugeneKyale** [comment]
@Camwyn I missed adding a change log entry last time round.

File: src/Tribe/Admin/Notice/Marketing/Stellar_Sale.php [hunk 2]
```diff
@@ -66,10 +66,19 @@ public function display_notice() {
...
+		$dependency   = tribe( \Tribe__Dependency::class );
 
 		ob_start();
-
-		include \Tribe__Main::instance()->plugin_path . 'src/admin-views/notices/tribe-stellar-sale.php';
+		if ( $dependency->has_active_premium_plugin() ) {
+			$this->get_template()->template( 'notices/tribe-stellar-sale-premium', $template_args, true );
+		} else {
+			$this->get_template()->template( 'notices/tribe-stellar-sale', $template_args, true );
+		}
```
Comment (bordoni) [inline src/Tribe/Admin/Notice/Marketing/Stellar_Sale.php:81]: Remove the `ob_*` methods here in favor of returning directly from template.

```suggestion
		if ( $dependency->has_active_premium_plugin() ) {
			return $this->get_template()->template( 'notices/tribe-stellar-sale-premium', $template_args, false );
		} else {
			return $this->get_template()->template( 'notices/tribe-stellar-sale', $template_args, false );
		}
```

**Review by bordoni — CHANGES_REQUESTED** [review]

**Review by bordoni — APPROVED** [review]

**Pull request merged**